ydb/library/yql/minikql/comp_nodes/mkql_wide_top_sort.cpp

```diff
@@ -601,38 +833,21 @@ using TBase = TComputationValue<TSpillingSupportState<Sort, HasCount>>;
     }
 
     void SealInMemory() {
-        if constexpr (!HasCount) {
-            static_assert (Sort);
-            // Remove placeholder for new data
-            Storage.resize(Storage.size() - Indexes.size());
-
-            Full.reserve(Storage.size() / Indexes.size());
-            for (auto it = Storage.begin(); it != Storage.end(); it += Indexes.size()) {
-                Full.emplace_back(&*it);
-            }
+        // Remove placeholder for new data
+        Storage.resize(Storage.size() - Indexes.size());
 
-            std::sort(Full.rbegin(), Full.rend(), LessFunc);
-            return;
+        Full.reserve(Storage.size() / Indexes.size());
+        for (auto it = Storage.begin(); it != Storage.end(); it += Indexes.size()) {
+            Full.emplace_back(&*it);
         }
 
-        Free.clear();
-        Free.shrink_to_fit();
-
-        if (Full.size() > Count) {
-            NYql::FastNthElement(Full.begin(), Full.begin() + Count, Full.end(), LessFunc);
-            Full.resize(Count);
-        }
-
-        if constexpr (Sort) {
-            std::sort(Full.rbegin(), Full.rend(), LessFunc);
-        }
+        std::sort(Full.rbegin(), Full.rend(), LessFunc);
     }
 
 public:
     EFetchResult InputStatus = EFetchResult::One;
-
+    NUdf::TUnboxedValuePod* Pointer = nullptr;
```

Comment:
Why do we need this `Pointer`?

Reply:
It's used in LLVM part here https://github.com/ydb-platform/ydb/pull/5314/files#diff-c09715a9b6990e43f38d93cf25307ea2da2434b595d97bccf90bb86c069de5adR1032 to write values direct their places.